crates/dap/src/client.rs

```diff
@@ -42,6 +44,8 @@ pub struct DebugAdapterClient {
     _process: Arc<Mutex<Option<Child>>>,
     sequence_count: AtomicU64,
     config: DebugAdapterConfig,
+    io_handlers: Arc<Mutex<Vec<IoHandler>>>,
+    log_io_handlers: Option<Arc<Mutex<Vec<IoHandler>>>>,
```

Comment:
I think the code would be simpler and more consistent with Zed's codebase if we create a `Model<io_handler>`. That way we can avoid management of the Mutex locks and let gpui handle it instead